src/network/gameclient.cc

```diff
@@ -249,12 +249,28 @@ void GameClient::run(std::unique_ptr<GameController>& ptr) {
 	d->modal = new FsMenu::LaunchMPG(capsule_, *this, *this, *this, *d->game, ptr, d->internet_);
 }
 
-void GameClient::do_run() {
+void GameClient::do_run(RecvPacket& packet) {
 	d->server_is_waiting = true;
 
 	Widelands::Game game;
 	game.set_write_syncstream(get_config_bool("write_syncstreams", true));
 
+	game.enabled_addons().clear();
+	for (size_t i = packet.unsigned_32(); i; --i) {
```

Comment:
As the packet format changes, the `NETWORK_PROTOCOL_VERSION` should be increased.